Providers/IntercomProvider.m

```diff
@@ -0,0 +1,133 @@
+//
+//  IntercomProvider.m
+
+#import "IntercomProvider.h"
+#import "Intercom.h"
+
+static NSString *const kIntercomSuperPropertiesKey = @"superProperties";
+
+@interface IntercomProvider ()
+
+@property(nonatomic, strong) NSDictionary *superProperties;
+
+@end
+
+
+@implementation IntercomProvider
+
+#ifdef AR_INTERCOM_EXISTS
+
+- (id)initWithApiKey:(NSString *)apiKey forAppId:(NSString *)appId {
+    [Intercom setApiKey:apiKey forAppId:appId];
+	self = [super init];
+    [self _unarchiveData];
+    return self;
+}
+
+- (void)event:(NSString *)event withProperties:(NSDictionary *)properties {
+	[Intercom logEventWithName:event metaData:properties];
+}
+
+- (void)didShowNewPageView:(NSString *)pageTitle withProperties:(NSDictionary *)properties {
+    NSString *event = [NSString stringWithFormat:@"View @%", pageTitle];
+    [Intercom logEventWithName:event metaData:properties];
+}
+
+- (void)identifyUserWithID:(NSString *)userID andEmailAddress:(NSString *)email {
+    [Intercom registerUserWithUserId:userID];
+    if (email) {
+         [Intercom updateUserWithAttributes:@{ @"email" : email}];
+    }
+}
+
+#pragma mark - Super Properties
+
+- (void)addSuperProperties:(NSDictionary *)properties {
+    properties = [properties copy];
+    [self _assertPropertyTypes:properties];
+    
+    NSMutableDictionary *mutableSuperProperties = [NSMutableDictionary dictionaryWithDictionary:self.superProperties];
+    [mutableSuperProperties addEntriesFromDictionary:properties];
+    self.superProperties = [mutableSuperProperties copy];
+    
+    [self _updateUserWithAttributes];
+    [self _archiveData];
+}
+
+- (void)removeSuperProperty:(NSString *)propertyName {
+    NSMutableDictionary *temp = [NSMutableDictionary dictionaryWithDictionary:self.superProperties];
+    [temp removeObjectForKey:propertyName];
+    self.superProperties = [temp copy];
+    
+    [self _updateUserWithAttributes];
+    [self _archiveData];
+}
+
+- (void)clearSuperProperties {
+    self.superProperties = @{};
+
+    [self _updateUserWithAttributes];
+    [self _archiveData];
+}
+
+#pragma mark - Private
+
+- (void)_archiveData {
+    NSString *filePath = [self _dataFilePath];
+    NSDictionary *data = @{kIntercomSuperPropertiesKey : self.superProperties};
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
I actually remember when this was a problem – even for private method, one place wanted the archiving to be synchronous to know when the process finished. Normally it is a good practise to have completion handler for async processes.
I guess it can wait until we run into problems™